d3a_api_client/redis_client_base.py

```diff
@@ -48,7 +47,7 @@ def __init__(self, area_id, client_id, autoregister=True, redis_url='redis://loc
     def _subscribe_to_response_channels(self, pubsub_thread=None):
         channel_subs = {
             self._response_topics[c]: self._generate_command_response_callback(c)
-            for c in Commands
+            for c in Commands if c in self._response_topics
```

Comment:
Can you please explain, why you dont want to subscribe to the response channels of the "new" commands e.g. UPDATE_BID ?

Reply:
Well it's not that I don't want to, I kept the existing callbacks and the main goal of the ticket was to remove the hardcoded strings from Commands class .
This is why I added the rest of commands to the enum and did not touch any logic in code
However, you are right ; I should probably add callbacks for the `update_offer`, `update_bid` , will do it

Reply:
why not the responses of all of the `Commands`?

Reply:
I'm not sure why these showed now , why are they implemented from the d3a side and not in here? I I thought that them not being implemented is because they are not needed ?
However, I really think that this got way beyond this ticket, as we need to create callbacks for several responses , add new commands ( Both the Rest and Redis Devices/markets ) 
I suggest creating another ticket to deal with this, what do you think ?

Reply:
You dont need to create new callbacks, all responses are handled in `_wait_and_consume_command_response`.

Reply:
Or rather `_publish_and_wait`

Reply:
@hannesdiedrich yes but I mean the methods that would need to call this function, like `def update_offer` and so on

Reply:
I see, you are right. But still I would vote for adding these in the frame of this story. @spyrostz What do you think?

Reply:
I'm in favor of not adding any response callback for the commands , as we have callbacks for events at this moment, and we have the on_event_or_response that can deal will all responses

Reply:
I agree with Hannes here, the responses are needed, and you cannot get away easily with handling this on _on_event_or_response. These commands are actually the synchronous mode of operation of our client, where the execution is blocked until the command response from the d3a is received. Therefore they are very useful if a user wants to wait until one action has finished. 
You could potentially wait for everything that arrives on _on_event_or_response and then unblock only the response that matters to you, however I see a disadvantage compared to subscribing to the response. _on_event_or_response is on a different thread than the typical caller of these methods (on_market_cycle or on_tick) , therefore you would have to introduce locks / polling, whereas in this case this is already handled and no new code needs to be written.

Reply:
@spyrostz great, one more thing- do you think we should tackle the mentioned feature in this PR or you advise creating a ticket for it?

Reply:
@Hassan754 it is not in the context of the ticket I agree, however it should not be a big change, and this PR has not bloated yet, therefore it will be relatively easy to review. IMO let's deal with it here in order for the issue to not get deprioritized/forgotten. Thanks for dealing with it !

Reply:
Deal 👌